test/e2e/network/ingress.go

```diff
@@ -596,6 +596,47 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
 			scaleAndValidateNEG(3)
 		})
 
+		It("should sync endpoints for both Ingress-referenced NEG and standalone NEG [Unreleased]", func() {
+			name := "hostname"
+			scaleAndValidateNEG := func(num int) {
+				scale, err := f.ClientSet.ExtensionsV1beta1().Deployments(ns).GetScale(name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				if scale.Spec.Replicas != int32(num) {
+					scale.Spec.Replicas = int32(num)
+					_, err = f.ClientSet.ExtensionsV1beta1().Deployments(ns).UpdateScale(name, scale)
+					Expect(err).NotTo(HaveOccurred())
+				}
+				wait.Poll(10*time.Second, framework.NEGUpdateTimeout, func() (bool, error) {
```

Comment:
This only checks the NEG used by ingress. You need another call to ListNetworkEndpoints of the other NEG thru exposed.